### PR TITLE
added blocktag logic

### LINKED ISSUE #50: Implement BlockTag logic

### PR DESCRIPTION
**Pull request by pabelnl**
closes #50

### CODE REVIEW

File: packages/aion/models/blocktag.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+const BLOCK_TAG = {
+    EARLIEST: "earliest",
+    LATEST: "latest"
+}
+
+class BlockTag {
+    constructor(block) {
+        this.defaultBlock = BLOCK_TAG.LATEST;
```
Comment (luyzdeleon) [inline packages/aion/models/blocktag.js]: Don't need this property.

File: packages/aion/models/blocktag.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+const BLOCK_TAG = {
+    EARLIEST: "earliest",
+    LATEST: "latest"
+}
+
+class BlockTag {
```
Comment (luyzdeleon) [inline packages/aion/models/blocktag.js:6]: Missing `toString` function

File: packages/aion/models/blocktag.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    constructor(block) {
+        this.defaultBlock = BLOCK_TAG.LATEST;
+        this.blockHeight = this.defaultBlock;
```
Comment (luyzdeleon) [inline packages/aion/models/blocktag.js]: Change this to: `this.blockHeight = BLOCK_TAG.LATEST`

File: packages/aion/models/blocktag.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        // Check if block is string type
+        if (typeof block == "string") {
+            if (this.hasZeroX(block)) {
+                this.blockHeight = block;
+            }else if(this.isValidBlockTag(block)){
+                this.blockHeight = BLOCK_TAG[block.toUpperCase()];
+            }
+        } else if (typeof block == "number") {// Check if block is type number
+            this.blockHeight = BigInt(block).toString(16);
+            this.blockHeight = this.prependZeroX(this.blockHeight);
```
Comment (luyzdeleon) [inline packages/aion/models/blocktag.js]: Can be done in one line: `this.blockHeight = this.prependZeroX(BigInt(block).toString(16));`

File: packages/aion/models/blocktag.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    prependZeroX(value){
+        if (this.hasZeroX(value)) {
+            return value;
+        }
+        return "0x"+value;
+    }
+    isValidBlockTag(value) {
+        value = value.toUpperCase();
+        // Check if value is a valid BLOCK_TAG
+        if (BLOCK_TAG[value]) {
```
Comment (luyzdeleon) [inline packages/aion/models/blocktag.js]: Can be done in one line: `BLOCK_TAG[value.toUpperCase()]`

**Review by luyzdeleon — CHANGES_REQUESTED** [review]

File: packages/aion/models/blocktag.js (hunk 2 above)
Comment (pabelnl) [reply packages/aion/models/blocktag.js:6]: oki

File: packages/aion/models/blocktag.js (hunk 4 above)
Comment (pabelnl) [reply packages/aion/models/blocktag.js]: yes, just didn't want a long line but I will change it.

File: packages/aion/models/blocktag.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+                if (this.hasZeroX(block)) {
+                    this.blockHeight = block;
+                }else if(this.isValidBlockTag(block)){
+                    this.blockHeight = BLOCK_TAG[block.toUpperCase()];
+                }
+            } else if (typeof block == "number") {// Check if block is type number
+                this.blockHeight = this.prependZeroX(BigInt(block).toString(16));
+            }
+        } catch (error) {
+            throw error;
```
Comment (luyzdeleon) [inline packages/aion/models/blocktag.js]: Don't throw, just set blockHeight to LATEST

**Review by luyzdeleon — CHANGES_REQUESTED** [review]

File: packages/aion/models/blocktag.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+                    this.blockHeight = block;
+                }else if(this.isValidBlockTag(block)){
+                    this.blockHeight = BLOCK_TAG[block.toUpperCase()];
+                }
+            } else if (typeof block == "number") {// Check if block is type number
+                this.blockHeight = this.prependZeroX(BigInt(block).toString(16));
+            }
+        } catch (error) {
+            this.blockHeight = BLOCK_TAG.LATEST;
+            console.log(error);
```
Comment (luyzdeleon) [inline packages/aion/models/blocktag.js]: `console.error`

**Review by luyzdeleon — CHANGES_REQUESTED** [review]

**Review by luyzdeleon — APPROVED** [review]

**Pull request merged**